server/src/main/java/org/opensearch/tracing/TracerManager.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.tracing.noop.NoopTracer;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+/**
+ * TracerManager represents a single global class that is used to access tracers.
+ *
+ * The Tracer singleton object can be retrieved using TracerManager.getTracer(). The TracerManager object
+ * is created during class initialization and cannot subsequently be changed.
+ */
+public class TracerManager {
+
+    private static final Logger logger = LogManager.getLogger(TracerManager.class);
+    private static volatile TracerManager INSTANCE;
+
+    private volatile Tracer defaultTracer;
+    private final Object mutex = new Object();
+    private final TracerSettings tracerSettings;
+    private final Supplier<Telemetry> telemetrySupplier;
+    private final ThreadPool threadPool;
+
+    /**
+     * Initializes the TracerFactory singleton instance
+     *
+     * @param tracerSettings       tracer settings instance
+     * @param threadPool           thread pool instance
+     */
+    public static synchronized void initTracerManager(
+        TracerSettings tracerSettings,
+        Supplier<Telemetry> tracerSupplier,
+        ThreadPool threadPool
+    ) {
+        if (INSTANCE == null) {
+            INSTANCE = new TracerManager(tracerSettings, tracerSupplier, threadPool);
+        } else {
+            logger.warn("Trying to double initialize TracerFactory, skipping");
+        }
+    }
+
+    /**
+     * Returns the {@link Tracer} singleton instance
+     * @return Tracer instance
+     */
+    public static Tracer getTracer() {
+        return INSTANCE == null ? NoopTracer.INSTANCE : INSTANCE.tracer();
+    }
+
```

Comment:
The singleton instance access isn't proper for instance if a consumer just wants to operate on a singleton instance it has to do a getTracer which can give it a `NoopTracer` if init has not been done which might not be expected. This can create unexpected bugs in future. The `getTracer ` should ensure it returns a concrete instance, probably creating one if it doesn't exist

Reply:
This code is removed now